src/node/collapser.ml

```diff
@@ -120,13 +128,19 @@ let collapse_until (type s) (tlog_coll:Tlogcollection.tlog_collection)
                     begin
                       maybe_log pi >>= fun () ->
                       begin
-                        if Sn.rem pi tlog_entries_per_file = 0L
+                        if pi = !next_rollover_point
                         then
-
-                          cb (Sn.of_int (first_tlog + !processed)) >>= fun () ->
-                          Lwt.return( processed := !processed + 1 )
+                          begin
+                            Logger.info_f_ "rollover_point pi:%Li next_rollover_point =%s"
+                                          pi (Sn.string_of !next_rollover_point)
+                            >>= fun () ->
+                            cb (first_tlog + !processed) >>= fun () ->
+                            let () = incr processed in
+                            let () = next_rollover_point := get_next_point (Sn.succ pi) in
```

Comment:
`let () = next_rollover_point := get_next_point pi in`
should work too? and stays correct even when you have a tlog with only 1 entry?

Also this got me thinking: are entries with the same i supposed to always be in the same tlog? If not what's the implication for the tlogmap and usages of # next_rollover ?